homeassistant/components/sunweg/sensor.py

```diff
@@ -0,0 +1,173 @@
+"""Read status of SunWEG inverters."""
+from __future__ import annotations
+
+import datetime
+import logging
+from types import MappingProxyType
+from typing import Any
+
+from sunweg.api import APIHelper
+from sunweg.device import Inverter
+from sunweg.plant import Plant
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import SunWEGData
+from .const import CONF_PLANT_ID, DEFAULT_PLANT_ID, DOMAIN
+from .sensor_types.inverter import INVERTER_SENSOR_TYPES
+from .sensor_types.phase import PHASE_SENSOR_TYPES
+from .sensor_types.sensor_entity_description import SunWEGSensorEntityDescription
+from .sensor_types.string import STRING_SENSOR_TYPES
+from .sensor_types.total import TOTAL_SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_device_list(api: APIHelper, config: MappingProxyType[str, Any]):
+    """Retrieve the device list for the selected plant."""
+    plant_id = config[CONF_PLANT_ID]
+
+    if plant_id == DEFAULT_PLANT_ID:
+        plant_info: list[Plant] = api.listPlants()
+        plant_id = plant_info[0].id
+
+    devices: list[Inverter] = []
+    # Get a list of devices for specified plant to add sensors for.
+    for inverter in api.plant(plant_id).inverters:
+        api.complete_inverter(inverter)
+        devices.append(inverter)
+    return [devices, plant_id]
```

Comment:
Since the length is static, a tuple makes more sense here

```suggestion
    return (devices, plant_id)
```